src/main/scala/com/fulcrumgenomics/util/PickIlluminaIndices.scala

```diff
@@ -0,0 +1,59 @@
+package com.fulcrumgenomics.util
+
+import scala.collection.JavaConversions._
+
+import com.fulcrumgenomics.cmdline.{ClpGroups, JeanLucTool}
+import dagr.commons.CommonsDef.{DirPath, FilePath}
+import dagr.sopt.{arg, clp}
+import picard.util.IlluminaUtil.IlluminaAdapterPair
+
+/**
+  * Program for picking sets of indices of arbitrary length that meet certain constraints
+  * and attempt to maximize the edit distance between all members of the set picked.
+  *
+  * @author Tim Fennell
+  */
+@clp(
+ description = "Picks a set of molecular indices that should work well together.",
+ group = classOf[ClpGroups.Utilities])
```

Comment:
This got me thinking.  I think there are two things we could do here, of which one would require a change to sopt.

1. Is much as you suggest, but it can't be `type` defs, it has to be `val`s.  So we could define a bunch of classes, and the have `ClpGroups` have a bunch of `val Utilities = classOf[UtilitiesGroup]`.
2. Would be a bigger change and effect `sopt`.  It turns out that java enums can implement interfaces - and so I think we could change group, instead of using a class, to using Java enums with:
```scala
trait Described { def description: String }
type Group[E <: Enum[E]] = java.lang.Enum[E] with Described
// and then in @clp
val group : Group[_]
```

I'm fairly sure this will allow the use of any enum that implements Described (name not final), so users of sopt can create their own enum types to represent groups, and use them.

Which do you prefer?